src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/ServerSentEventsMessageParser.cs

```diff
@@ -36,28 +36,15 @@ public ParseResult ParseMessage(ReadOnlySequence<byte> buffer, out SequencePosit
         {
             if (!(buffer.PositionOf(ByteLF) is SequencePosition lineEnd))
             {
-                // For the case of data: Foo\r\n\r\<Anything except \n>
-                if (_internalParserState == InternalParseState.ReadEndOfMessage)
-                {
-                    if (ConvertBufferToSpan(buffer.Slice(start, buffer.End)).Length > 1)
-                    {
-                        throw new FormatException("Expected a \\r\\n frame ending");
-                    }
-                }
-
                 // Partial message. We need to read more.
                 return ParseResult.Incomplete;
             }
 
+            // buffer, and thus line should atleast contain \n at this point.
             lineEnd = buffer.GetPosition(1, lineEnd);
             var line = ConvertBufferToSpan(buffer.Slice(start, lineEnd));
             buffer = buffer.Slice(line.Length);
 
-            if (line.Length <= 1)
```

Comment:
This was removed as we can't really get to `0` chars, due to line being always atleast `\n`